src/de.rs

```diff
@@ -701,8 +701,17 @@ impl<'de, 'a> de::Deserializer<'de> for &'a mut Deserializer<'de> {
     where
         V: Visitor<'de>,
     {
-        // Ignore extra fields/options
-        visitor.visit_none()
+        match self.peek_major()? {
+            0 => self.deserialize_u32(visitor),
+            1 => self.deserialize_i32(visitor),
```

Comment:
This means that a value that does not fit in a `i32` or a `u32` de-serialization will fail.
I believe this should not actually de-serialize the value but just discard it.